### ISSUE TITLE
details about UniAD

### ISSUE THREAD

**Issue by angela00010** [xcyao00/ResAD#4]
hi, thank you for your interesting work! 
1. I reproduce your code, but can not reach the result in paper, especially in mvtec_to_visa 4 shot setting, I get AUROC for 75% and 87% in img and pixel level. I did't use your refence image but random select 4 shot, is this the key point?  By the way, can you open some pretrained models?
 
2. I found UniAD + res_features improve result a lot, is this mean simply change ori_feature to res_feature? or have to add some other loss?  I cannot get more details, can you open these code ?

can you give me some advices? thank you!

**Comment by xcyao00** [comment]
You can try the samples I provided and see if can reach the results. I don't think just changing the samples will have that much impact on the results. Sorry, I didn't save the trained models. But before open-souring the code, I ran it once and was able to achieve the results I wrote in README.md. Perhaps you can run more epochs.

For UniAD, I just replace the orginal features to residual features, no other losses.

**Comment by angela00010** [comment]
> You can try the samples I provided and see if can reach the results. I don't think just changing the samples will have that much impact on the results. Sorry, I didn't save the trained models. But before open-souring the code, I ran it once and was able to achieve the results I wrote in README.md. Perhaps you can run more epochs.
> 
> For UniAD, I just replace the orginal features to residual features, no other losses.

thank you for your reply! I notice residual features in paper means directly plus operatio. This is also different from InCTRL. But in code I see MSE feature map works (pos_flag=True). So which one would you recommend ?

**Comment by Diana1026** [comment]
> > You can try the samples I provided and see if can reach the results. I don't think just changing the samples will have that much impact on the results. Sorry, I didn't save the trained models. But before open-souring the code, I ran it once and was able to achieve the results I wrote in README.md. Perhaps you can run more epochs.
> > For UniAD, I just replace the orginal features to residual features, no other losses.
> 
> Thank you for your reply! I notice residual features in paper means subtraction operation. This is also different from InCTRL. But in code I see MSE feature map works (pos_flag=True). So which one would you recommend ?

Hi, may I know if you reproduced the 4-shot results when using the provided reference images?
